lanelet2/lanelet2_routing/include/lanelet2_routing/Route.h

```diff
@@ -196,10 +196,22 @@ class Route {
    *  @return Vector of errors if no-throw is chosen and errors are found */
   Errors checkValidity(bool throwOnError = false) const;
 
+  /** @brief Returns the endpoint location at the end of the route.
+   *         Guaranteed to be on the last lanelet of the route and middlepoint of last left and right points by default
+   *         Can be changed later by setEndPoint() 
+   *  @return Endpoint location at the end of the route. */
+  lanelet::Point3d getEndPoint() const;
+
+  /** @brief Set the endpoint location at the end of the route.
+   *  @param end_point that is within the last lanelet of the route
+   *  @throw Throws invalid_argument if the point is not within the last lanelet's boundary */
+  void setEndPoint(lanelet::Point3d end_point);
+
  private:
   std::unique_ptr<internal::RouteGraph> graph_;  ///< The internal graph
   LaneletPath shortestPath_;                     ///< The underlying shortest path used to create the route
   LaneletSubmapConstPtr laneletSubmap_;          ///< LaneletSubmap with all lanelets that are part of the route
+  lanelet::Point3d end_point_;              ///< Endpoint of route
```

Comment:
How is this point being communicated to carma_wm? I think we need to include it in the cav_msgs/Route.msg in order for it to be set in carma_wm

Reply:
added it to route package and wm_listener's routeCallback

Reply:
@MishkaMN Doesn't it need to be sent in the cav_msgs/Route though? That message is not a binary serialization like the map. It's a list of lanelets, so I think we need the 3d end point so that wm_listener can set it properly. Is there a PR missing?